lib/boardgamegeek_client.ex

```diff
@@ -5,59 +5,64 @@ defmodule BoardGameGeekClient do
 
   require BoardGameGeek
 
-  def get_game_collection(username, wait \\ 500) do
-    url = "collection?username=#{username}&own=1&subtype=boardgame"
-    doc = get_response(url, wait)
-    game_ids = Exml.get doc, "//items/item/@objectid"
-    Process.sleep(wait)
-    get_games_info(game_ids)
+  def search_games(query) do
+    url = "search?query=#{query}&type=boardgame"
+    doc = get_response(url)
+    ids = Exml.get(doc, "//items/item/@id")
+    names = Exml.get(doc, "//items/item[@id]/name/@value")
+    Enum.zip(ids, names) |> Enum.map(fn {k, v} -> %{id: k, name: v} end)
   end
 
-  def get_games_info(game_ids, wait \\ 500) when is_list(game_ids), do: get_games_info(game_ids, wait, [])
-  defp get_games_info([], _wait, acc), do: acc
-  defp get_games_info([head|tail], wait, acc) do
-    url = "thing?id=#{head}"
-    response = BoardGameGeek.post(url)
-    case response.status_code do
-      202 ->
-        Process.sleep(wait)
-        # Results not ready, append to end of games list so that it retries
-        get_games_info(tail ++ [head], wait, acc)
-      200 ->
-        Process.sleep(wait)
-        get_games_info(tail, wait, [game_from_response(response) | acc])
-      true -> raise "unknown status code: #{response.status_code}"
-    end
+  def get_game_collection(username) do
+    game_ids =
+      "collection?username=#{username}&own=1&subtype=boardgame"
+      |> get_response
+      |> Exml.get("//items/item/@objectid")
+    get_games_info(game_ids)
   end
 
-  defp game_from_response(response) do
-    response.body
-    |> Exml.parse
-    |> game_from_xml
+  def get_games_info(game_ids) when is_list(game_ids) do
+    doc = "thing?id=#{Enum.join(game_ids, ",")}"
+    |> get_response(10_000)
+
+    case games_from_xml(doc) do
```

Comment:
Looking for some feedback here. This feels unnecessary, but it also seemed like a good idea to return a tuple with either `:ok` or `:error`